docker/README.md

```diff
@@ -0,0 +1,81 @@
+# Garage Docker
+
+Currently there's three versions of garage containers:
+  - ci: used for the continuous integration of garage in TravisCI and not
+    intended for final users.
+  - release: it's a published image based on the latest official release of
+    garage.
+  - local: it's an image built on top of the release with a copy of the files
```

Comment:
this is not called .local, it's just called the Dockerfile. 

whereas we put special Dockerfiles in docker/, the canonical Dockerfile usually goes in the repository root. We might move all Dockerfiles to the repo root--not sure yet. That's out of scope for this PR.